### PR TITLE
API: Split internal instrument.base.py module

### PR DESCRIPTION
**Pull request by jenshnielsen**
Split base into instruument.py and instrument_base.py 

This matches how the parameters module now does it and has the advantage that 

1. No more ambiguous base.py 
2. Easier to work with the Instrumentbase and Instrument classes

### CODE REVIEW

**Comment by jenshnielsen** [comment]
To be merged after https://github.com/QCoDeS/Qcodes/pull/4281 to not create a conflict

File: qcodes/tests/parameter/test_parameter_basics.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import pytest
 
 import qcodes.utils.validators as vals
-from qcodes.instrument.function import Function
+from qcodes.instrument import Function
```
Comment (jenshnielsen) [inline qcodes/tests/parameter/test_parameter_basics.py:4]: It's starting to smell like Functions should be in parameters module ? (not for this pr thou)

Comment (jenshnielsen) [reply qcodes/tests/parameter/test_parameter_basics.py:4]: Done as part of https://github.com/QCoDeS/Qcodes/pull/4286

File: qcodes/tests/test_monitor.py [hunk 2]
```diff
@@ -8,8 +8,8 @@
 import pytest
 import websockets
 
-from qcodes.instrument.base import Parameter
```
[the remark this answers is not in the document]
Comment (astafan8) [reply qcodes/tests/test_monitor.py:11]: > we should add it there for backwards compatibility

probably :(

File: qcodes/tests/test_instrument.py [hunk 3]
```diff
@@ -8,8 +8,12 @@
 
 import pytest
 
-from qcodes.instrument.base import Instrument, InstrumentBase, find_or_create_instrument
-from qcodes.instrument.function import Function
+from qcodes.instrument import (
+    Function,
```
Comment (jenshnielsen) [inline qcodes/tests/test_instrument.py:12]: Update import once https://github.com/QCoDeS/Qcodes/pull/4286 lands

File: qcodes/instrument/instrument_base.py [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+)
+
+import numpy as np
+
+from qcodes.logger.instrument_logger import get_instrument_logger
+from qcodes.parameters import Parameter, ParameterBase
+from qcodes.utils.helpers import DelegateAttributes, full_class
+from qcodes.utils.metadata import Metadatable
+
+from .function import Function
```
Comment (astafan8) [inline qcodes/instrument/instrument_base.py:24]: import from .parameters?

File: qcodes/instrument/ip_to_visa.py [hunk 5]
```diff
@@ -1,16 +1,16 @@
-from typing import Optional, Any
+from typing import Any, Optional
 
-from qcodes.instrument.base import Instrument
+import qcodes.utils.validators as vals
+from qcodes.instrument.instrument import Instrument
 from qcodes.instrument.ip import IPInstrument
```
Comment (astafan8) [inline qcodes/instrument/ip_to_visa.py:5]:
```suggestion
from .instrument import Instrument
from .ip import IPInstrument
```

File: qcodes/instrument/ip_to_visa.py [hunk 6]
```diff
@@ -1,16 +1,16 @@
...
+
 # previous to introducing the `InstrumentLoggerAdapter` the IPToVisa instrument
 # was logging in the name of the `VisaInstrument`. To maintain that behaviour
 # import the `instrument.visa.log` and log to this one.
-from qcodes.instrument.visa import VisaInstrument, VISA_LOGGER
+from qcodes.instrument.visa import VISA_LOGGER, VisaInstrument
```
Comment (astafan8) [inline qcodes/instrument/ip_to_visa.py:10]:
```suggestion
from .visa import VISA_LOGGER, VisaInstrument
```

**Review by astafan8 — APPROVED** [review]

**Comment by jenshnielsen** [comment]
bors merge

**Pull request merged**